src/control/cad.js

```diff
@@ -251,7 +251,7 @@ class CadControl extends Control {
    */
   getClosestFeatures(coordinate, numFeatures) {
     const num = numFeatures || 1;
-    const ext = [-Infinity, -Infinity, Infinity, Infinity];
+    const ext = this.map.getView().calculateExtent();
```

Comment:
I can see different use case here. I propose to add a property which take a function. I do it master. Thanks for pointing this out.